checkpoint-interface.js

```diff
@@ -36,11 +36,15 @@ function checkpointInterface (trie) {
  */
 function checkpoint () {
   var self = this
-  var wasCheckpoint = self.isCheckpoint
-  self._checkpoints.push(self.root)
-  if (!wasCheckpoint && self.isCheckpoint) {
-    self._enterCpMode()
-  }
+  self.sem.take(function () {
```

Comment:
`checkpoint` used to be synchronous, but this change makes it asynchronous. As a result, this function should accept a callback argument.

Reply:
You're right. I had it on the list but then missed it. Unfortunately this fix is not legit without the callback as it defers the commit and it also unlikely to work without getting the semaphore there as enterCpMode mutates the db arrays for get and  puts. 
I'll look at it this evening and check where it gets called. I'll close this for now